File: kotlin/hamming/bin/main/Hamming.kt
class Hamming {
    companion object {
        fun compute(left: String, right: String): Int {
            require(left.length == right.length) { 
                "left and right strands must be of equal length." 
            }
            return (left zip right).count { it.first != it.second }
        }
    }
}

File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        //if (firstStrand.Length == 0 && secondStrand.Length == 0) return 0;
        if (firstStrand.Length != secondStrand.Length) throw new ArgumentException();
        //if (firstStrand.Length == 0 || secondStrand.Length == 0) throw new ArgumentException();
        return firstStrand.Zip(secondStrand, (first, second) => (first == second) ? 0 : 1).Sum();

    }
}